src/components/PieChart.js

```diff
@@ -0,0 +1,182 @@
+const React = require('react');
+const ReactDOM = require('react-dom');
+const d3 = require('d3');
+
+const StyleConstants = require('../constants/Style');
+
+const PieChart = React.createClass({
+  propTypes: {
+    animateDuration: React.PropTypes.number,
+    animationTypeOnLoad: React.PropTypes.oneOf(['roll', 'pop']),
+    arcWidth: React.PropTypes.number,
+    colors: React.PropTypes.array,
+    data: React.PropTypes.array.isRequired,
+    height: React.PropTypes.number,
+    hoverExpandDistance: React.PropTypes.number,
+    onLabelClick: React.PropTypes.func,
+    onMouseEnter: React.PropTypes.func,
+    onMouseLeave: React.PropTypes.func,
+    onPieClick: React.PropTypes.func,
+    showDataLabel: React.PropTypes.bool,
+    style: React.PropTypes.object,
+    width: React.PropTypes.number
+  },
+
+  getDefaultProps () {
+    return {
+      animationTypeOnLoad: 'roll',
+      animateDuration: 500,
+      arcWidth: 2,
+      colors: [StyleConstants.Colors.PRIMARY].concat(d3.scale.category20().range()),
+      height: 350,
+      hoverExpandDistance: 10,
+      onLabelClick () {},
+      onMouseEnter () {},
+      onMouseLeave () {},
+      onPieClick () {},
+      showDataLabel: true,
+      width: 300
+    };
+  },
+
+  componentDidMount () {
+    const dom = ReactDOM.findDOMNode(this.refs.chart);
+
+    this._renderChart(dom);
+  },
```

Comment:
@cerinman You are correct. The chart won't update because it is already mounted. I'm not sure if there is any situation we would change the chart data on the fly. In the Investment widget, we don't change the chart data once it is mounted. Do you have any recommendations on how we could update the chart as its data changes and still keep the animations?